challangeNetflix/MovieDetails/MovieDetailsInteractor.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  MovieDetailsInteractor.swift
+//  challangeNetflix
+//
+//  Created by Larissa Diniz  on 12/05/20.
+//  Copyright © 2020 Larissa Diniz . All rights reserved.
+//
+
+import Foundation
+import RealmSwift
+
+protocol MovieDetailsInteractorProtocol: class {
+    func onViewLoaded()
+    func onFavTapped()
+    func showTrailler()
+}
+
+class MovieDetailsInteractor: MovieDetailsInteractorProtocol {
+    
+    private let presenter: MovieDetailsPresentable
+    private let worker: MovieDetailsWorkerProtocol
+    
+    init(presenter: MovieDetailsPresentable, worker: MovieDetailsWorkerProtocol) {
+        self.presenter = presenter
+        self.worker = worker
+    }
+    
+    func onViewLoaded() {
+        self.presenter.show(item: worker.movie)
+        let retrievedObject = worker.retrievedObject()
+        if retrievedObject != nil {
+            presenter.toggleFavButton(favorite: true)
+        } else {
+            presenter.toggleFavButton(favorite: false)
+        }
+    }
+    
+    func onFavTapped() {
+        let retrievedObject = worker.retrievedObject()
+        if retrievedObject != nil {
+            worker.delete(object: retrievedObject!)
+            presenter.toggleFavButton(favorite: false)
+        } else {
+            worker.add()
```

Comment:
worker.add() ? Não fica claro o que essa função ta fazendo

Reply:
era addDatabase mas o rafa pediu pra tirar

Reply:
Mas worker.add() tambem nao diz nada... eu nao sei o que ta fazendo essa função... O que ta sendo adicionado nesse add? se fosse worker.addDatabase() tambem nao faria sentido, iria parecer que tu ta adicionando um banco de dados

Reply:
agora eu deixei add(movie: Movie), assim da de entender que to adicionando um filme né?